include/slang/syntax/SyntaxNode.h

```diff
@@ -277,6 +310,9 @@ class SLANG_EXPORT SeparatedSyntaxList : public SyntaxListBase {
     SeparatedSyntaxList(span<TokenOrSyntax> elements) :
         SyntaxListBase(SyntaxKind::SeparatedList, elements.size()), elements(elements) {}
 
+    /// @return the elements of nodes in the list
+    [[nodiscard]] span<TokenOrSyntax> elems() const { return elements; }
```

Comment:
I don't think this should return a span<TokenOrSyntax> if elems() is const, since that would allow modification of the underlying elements. I think there should be two overloads, the const one should return a span<const ConstTokenOrSyntax> and the non-const one can return what you have here.